### PR TITLE
jobs/build: Add NO_UPLOAD parameter to allow for debugging

### PR DESCRIPTION
**Pull request by dustymabe**
This allows for in pipeline debugging without worrying about pushing
artifacts over multiple debug runs.

### CODE REVIEW

File: jobs/build.Jenkinsfile [hunk 1]
```diff
@@ -63,6 +63,9 @@ properties([
       booleanParam(name: 'KOLA_RUN_SLEEP',
                    defaultValue: false,
                    description: 'Just wait forever when we get to running kola tests (temporary)'),
```
Comment (jlebon) [inline jobs/build.Jenkinsfile:65]: Maybe let's add a "Implies NO_UPLOAD." here?

**Review by jlebon — COMMENTED** [review]
We need to also gate multi-arch triggering (or better, add a `NO_UPLOAD` parameter to that job too and pass it through).

We need to also gate publishing.

File: jobs/build.Jenkinsfile (hunk 1 above)
Comment (dustymabe) [reply jobs/build.Jenkinsfile:65]: done

File: jobs/build.Jenkinsfile [hunk 2]
```diff
@@ -360,11 +371,11 @@ lock(resource: "build-${params.STREAM}") {
...
 
-        // Do an Early Archive of just the OSTree. This has the
-        // desired side effect of reserving our build ID before
-        // we fork off multi-arch builds.
+        // If we are uploading results then let's do an early archive
+        // of just the OSTree. This has the desired side effect of
+        // reserving our build ID before we fork off multi-arch builds.
         stage('Archive OSTree') {
-            if (s3_stream_dir) {
+            if (uploading) {
```
[the remark this answers is not in the document]
Comment (dustymabe) [reply jobs/build.Jenkinsfile:378]: yeah, there are enough of these where I'd prefer a separate PR if we go that route.

**Comment by dustymabe** [comment]
> We need to also gate multi-arch triggering (or better, add a `NO_UPLOAD` parameter to that job too and pass it through).

For now I'll just gate it here and maybe add the `NO_UPLOAD` parameter to the multi-arch job when we unify `build` and `build-arch` here soon.

> We need to also gate publishing.

done

**Review by jlebon — APPROVED** [review]

**Comment by jlebon** [comment]
I like the replacement of `s3_stream_dir` by `uploading`. Much clearer!

**Pull request merged**